packages/api/internal/orchestrator/node.go

```diff
@@ -236,8 +282,10 @@ func (n *Node) InsertBuild(buildID string) {
 	n.buildCache.Set(buildID, struct{}{}, 2*time.Minute)
 }
 
-func (o *Orchestrator) NodeCount() int {
-	return o.nodes.Count()
+func (n *Node) GetClient() (*grpclient.GRPCClient, RequestContextBuilder) {
```

Comment:
can we set the metadata per client to be injected on every request automatically?

Reply:
Looks like it cannot be done. It cannot be set on the gRPC client or gRPC connection directly. Basically, to put it in, we need to adjust ctx provided during method invoke.